wikicurses/main.py

```diff
@@ -35,6 +37,111 @@ def tabComplete(text, matches):
     return match
 
 
+def showImages():
+    """
+    Launch program to display the images for the current article.
+
+    The program is specified in the config, along with it's args.
+    Otherwise, the default is `feh`.
+    """
+    if 'wikipedia.org' not in wiki.siteurl:
+        ex.notify('Cancelled: Only works on Wikipedia')
+        return
+
+    targets = fetchImageTargets(page.title)
+
+    if not targets:
+        ex.notify('No relevant images found')
+        return
+
+    ## We make a copy of the returned list here,
+    ## because it is memoized with `lru_cache` and we will be deleting
+    ## elements from it non-permanently.
+    image_info = list(fetchImageInfo(page.title))
+    filtered = []
+
+    for target in targets:
+        for index, i in enumerate(image_info):
+            title, url = i
+            if target in title:
+                filtered.append(url)
+                del image_info[index] # prevent more than one occurrence of image
+                break
+
+    try:
+        command = [settings.conf.get('images', 'program')]
+    except (settings.configparser.NoOptionError,
+            settings.configparser.NoSectionError):
+        command = ['feh']
+
+    try:
+        args = settings.conf.get('images', 'arguments').split(' ')
+    except (settings.configparser.NoOptionError,
+            settings.configparser.NoSectionError):
+        if command == ['feh']:
+            args = ['-q', '--scale-down', '--image-bg', 'white', '-g', '400x400']
+        else:
+            args = []
+
+    command.extend(args)
+    command.extend(filtered)
+
+    try:
+        with open(os.devnull, 'w') as fp:
+            subprocess.Popen(command, stdout=fp, stderr=fp)
+    except FileNotFoundError:
+        ex.notify('Program not found')
+        return
+
+
+@lru_cache()
+def fetchImageTargets(page_title):
+    """
+    Get filenames of relevant images from the current article.
+
+    These are used to filter out non-relevant images from the API result.
+    """
+    url = re.sub(r'api.php', 'index.php', wiki.siteurl)
+    page_title = re.sub(r' ', '_', page_title)
+    raw = wiki._query(customurl=url, action="raw", title=page_title)
+
+    ### Finding targets...
+    targets = []
+
+    ## of the form `[[Image:foobar.png]]`
+    for match in re.finditer(r'\b(?:File:|Image:)([^]|\n\r]+)', raw):
```

Comment:
Ah! I see. I'm still new to MediaWiki in general. Hm, well the case issue is easy, but I don't want to resolve the local aliases issue with another query. I'll have to think about a relatively foolproof solution to that.